fastlane/lanes/build.rb

```diff
@@ -359,4 +367,46 @@ def generate_prototype_build_number
       "#{branch}-#{commit}"
     end
   end
+
+  # Uploads the React Native JavaScript bundle and source map files.
+  # These files are provided by the Gutenberg Mobile library.
+  #
+  # @param [String] app App name, e.g. 'WordPress' or 'Jetpack'.
+  # @param [String] flavor Build flavor, e.g. 'Jalapeno' or 'Vanilla'.
+  # @param [String] build_type Build type, e.g. 'Debug' or 'Release'.
+  # @param [String] release_version Release version name to attach the files to in Sentry.
+  #
+  def upload_gutenberg_sourcemaps(app:, flavor:, build_type:, release_version:)
+    # Load Sentry properties
+    sentry_path = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'src', app, 'sentry.properties')
+    sentry_properties = JavaProperties.load(sentry_path)
+    sentry_token = sentry_properties[:'auth.token']
+    project_slug = sentry_properties[:'defaults.project']
+    org_slug = sentry_properties[:'defaults.org']
+
+    # The bundle and source map files are extracted from merged assets location created after building the app.
+    # The format is: <app><flavor><build_type>
+    # E.g.: jetpackJalapenoDebug
+    build_asset_folder_name = "#{app.downcase}#{flavor}#{build_type}"
+    bundle_source_map_path = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'build', 'intermediates', 'assets', build_asset_folder_name)
+
+    Dir.mktmpdir do |sourcemaps_folder|
+      # It's important that the bundle and source map files have specific names, otherwise, Sentry won't symbolicate the stack traces.
+      FileUtils.cp(File.join(bundle_source_map_path, 'index.android.bundle'), File.join(sourcemaps_folder, 'index.android.bundle'))
+      FileUtils.cp(File.join(bundle_source_map_path, 'index.android.bundle.map'), File.join(sourcemaps_folder, 'index.android.bundle.map'))
```

Comment:
Any particular reason why you copy those files in a temp folder (while the files you copy seems to already have the right file names, so that's apparently not for that reason)? As opposed to directly passing `sourcemap: bundle_source_map_path` to the `sentry_upload_sourcemap` action?

Or is it because you fear there might be more sourcemap files in that `bundle_source_map_path` folder that would be accidentally uploaded if so and that's why you're isolating just those two in a temp folder?

Reply:
Good observation @AliSoftware. I copied this function from iOS counterpart (https://github.com/wordpress-mobile/WordPress-iOS/pull/22655/files#diff-72b7ec258b7b79c415642e3cb618b99665e0d8e491e4b8aafe468a88636c486fR489-R493), in that case, the files need to be renamed so makes sense to copy them to a temporary folder. As you pointed out, on Android since they have the expected filenames we could directly use the original files. However, I left the copy step for the reason you mentioned about isolating the files just in case there are other sourcemap files. I'd like to note that currently there aren't others, but since the `build_asset_folder_name` contains all merged assets, this action would help us to avoid issues in the future.